src/error.rs

```diff
@@ -72,4 +72,8 @@ pub enum Error {
     ConnectionSendError,
     #[error("failed to send initial TransportEvent::NewAddress")]
     SendErrorTransportEvent,
+    #[error("dialing address timed out")]
+    DialTimeout(#[from] tokio::time::error::Elapsed),
+    #[error("failed to establish network connection {0:?}")]
+    ConnectionTimeout(std::time::Duration),
```

Comment:
in this case it's specifically for streams, so I would rename this `StreamOpenTimeout` or something like that!